### ISSUE TITLE
$this->User->logout does not show logout link anywhere except on Application root

### ISSUE THREAD

**Issue by Juergen-Airscout** [CakeDC/users#651]
I am working on implementing the CakeDC's Users plugin in my application and was testing out the UserHelper. I tried to generate a logout link but was unable to get "$this->User->logout('logout')" to work. I then went on to try the welcome function and it would only show "Welcome, ". On a whim, I then went to the root of the application and suddenly the welcome link showed the full "Welcome, <current logged in user as a profile link>". In addition to that, the logout link was also working. I've tried this as a superuser and as a standard user all to the same effect. Any ideas what might be happening? Unsure if it is related to #538 or #469 but since a patch was released for that issue i don't imagine so. Any ideas what might be happening?

**Comment by juerujin** [comment]
I have found the answer and so I will put it here so others may benefit. Turn out that although I put the `$this->loadComponent('CakeDC/Users.UsersAuth');` in my AppController.php, I did not have `parent::initialize();` in my sub-controllers. It hadn't proved to a problem up until this point and so I hadn't noticed them missing. As a result, the components of the parents were not being loaded. That single line of code has solved 3 days of problems. Not sure how to close an issue so I'll have to wait for an admin to close this.
